airbyte-db/lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_39_1_001__CreateStreamReset.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.configs.migrations;
+
+import static org.jooq.impl.DSL.currentOffsetDateTime;
+import static org.jooq.impl.DSL.unique;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
+public class V0_39_1_001__CreateStreamReset extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V0_39_1_001__CreateStreamReset.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    final DSLContext ctx = DSL.using(context.getConnection());
+    createStreamResetTable(ctx);
+  }
+
+  private static void createStreamResetTable(final DSLContext ctx) {
+    final Field<UUID> id = DSL.field("id", SQLDataType.UUID.nullable(false));
+    final Field<UUID> connectionId = DSL.field("connection_id", SQLDataType.UUID.nullable(false));
+    final Field<String> streamNamespace = DSL.field("stream_namespace", SQLDataType.CLOB.nullable(false));
+    final Field<String> streamName = DSL.field("stream_name", SQLDataType.CLOB.nullable(false));
+    final Field<OffsetDateTime> createdAt =
+        DSL.field("created_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false).defaultValue(currentOffsetDateTime()));
+    final Field<OffsetDateTime> updatedAt =
+        DSL.field("updated_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false).defaultValue(currentOffsetDateTime()));
+
+    ctx.createTableIfNotExists("stream_reset")
+        .columns(id, connectionId, streamNamespace, streamName, createdAt, updatedAt)
+        .constraints(
+            unique(connectionId, streamNamespace, streamName))
+        .execute();
+
+    ctx.createIndex("connection_id_stream_namespace_name_idx").on("stream_reset", "connection_id", "stream_namespace", "stream_name").execute();
```

Comment:
For both the `unique` constraint and the index, I think we should order it as `connectionId, streamName, streamNamespace`, because as I mentioned above its possible for the namespace to be null

Reply:
huh. why does this matter? i had never taken this into consideration when building an index.

Reply:
I was thinking if there was a connection that did not use namespaces, then the queries on this table for that connection would be based only on `connectionId` and `streamName`. Because of this, I thought we would want this index to be ordered as `connectionId, streamName, streamNamespace`, so that queries that are only specifying `where` clauses on `connectionId` and `streamName` would still be able to take full advantage of the index